src/parser.rs

```diff
@@ -375,6 +267,35 @@ pub fn check_cpu_support() {
     }
 }
 
+#[inline(always)]
+fn parse_coordinate(buffer: *const u8, len: &mut usize) -> (usize, bool) {
```

Comment:
Could we please rename `len` to `buffer_index` (or similar), as `len` is a bit confusing?
Same goes for `parse_pixel_coordinates`

Reply:
it used to be the length, until I realized that managing the length separately and then adding it to the index is slower than outright modifying the index. And  I never cleaned that bit up.